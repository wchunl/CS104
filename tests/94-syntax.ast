ROOT "" (0.0.0)
|   TOK_RETURN "return" (4.5.1)
|   |   TOK_IDENT "foo" (4.5.8) variable
