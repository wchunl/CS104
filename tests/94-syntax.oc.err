h¨:4.9: syntax error, unexpected ';'
h¨:6.8: syntax error, unexpected TOK_IDENT
h¨:7.10: invalid source character (.)
