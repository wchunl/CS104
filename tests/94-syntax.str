string_set[   2]:   11868120863286193964 0x200b118->"main"
                    3930163824820145617 0x200b0c8->"static"
string_set[   3]:    7563291569981072216 0x200b168->"String"
                   11597697714117577063 0x200add8->"="
                    5344662657899890615 0x200ae88->";"
string_set[   6]:   14635384904416252381 0x200aed8->"return"
string_set[  11]:      21606623943142546 0x200ac18->"{"
string_set[  12]:   13686721134187162920 0x200b398->"}"
                   13581035385482130682 0x200b1b8->"["
string_set[  13]:    6895825537633130899 0x200b258->"args"
string_set[  14]:   15816506932648033945 0x200aa88->"("
                    6142509188972423790 0x2006738->""
                   12868445110721718657 0x200ab38->")"
string_set[  15]:   12051031859636400188 0x200b208->"]"
string_set[  16]:   11752052362951903271 0x200b078->"public"
string_set[  17]:    8968846175329310707 0x2006758->"f"
                    4993892634952068459 0x200ad28->"a"
string_set[  18]:   16711936661946036647 0x200b2f8->"exit"
string_set[  20]:    9631199822919835226 0x200af28->"foo"
string_set[  21]:    8194355264956552874 0x200b348->"255"
                   14032891358763643253 0x200b2a8->"System"
                   12597418120160561823 0x2006318->"int"
load_factor = 0.957
bucket_count = 23
max_bucket_size = 3
