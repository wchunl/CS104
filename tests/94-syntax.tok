#  1 "94-syntax.oc"
#  1 "<built-in>"
#  1 "<command-line>"
#  1 "94-syntax.oc"
   4 3.00001 262 TOK_INT       int
   4 3.00005 282 TOK_IDENT     f
   4 3.00006  40 '('           (
   4 3.00007  41 ')'           )
   4 3.00009 123 '{'           {
   4 4.00001 262 TOK_INT       int
   4 4.00005 282 TOK_IDENT     a
   4 4.00007  61 '='           =
   4 4.00009  59 ';'           ;
   4 5.00001 268 TOK_RETURN    return
   4 5.00008 282 TOK_IDENT     foo
   4 5.00011  59 ';'           ;
   4 6.00001 282 TOK_IDENT     public
   4 6.00008 282 TOK_IDENT     static
   4 6.00015 262 TOK_INT       int
   4 6.00019 282 TOK_IDENT     main
   4 6.00024  40 '('           (
   4 6.00025 282 TOK_IDENT     String
   4 6.00031  91 '['           [
   4 6.00032  93 ']'           ]
   4 6.00034 282 TOK_IDENT     args
   4 6.00038  41 ')'           )
   4 6.00040 123 '{'           {
   4 7.00004 282 TOK_IDENT     System
   4 7.00011 282 TOK_IDENT     exit
   4 7.00016  40 '('           (
   4 7.00017 283 TOK_INTCON    255
   4 7.00020  41 ')'           )
   4 7.00021  59 ';'           ;
   4 8.00001 125 '}'           }
