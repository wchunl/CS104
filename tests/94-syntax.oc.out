; 0.  1: # 1 "94-syntax.oc"; 1.  0: 
; 1.  1: # 1 "<built-in>"
; 2.  1: # 1 "<command-line>"
; 3.  1: # 1 "94-syntax.oc"
; 4.  1: 
; 4.  2: 
; 4.  3: int f() {
; 4.  4: int a = ;
; 4.  5: return foo;
; 4.  6: public static int main (String[] args) {
; 4.  7:    System.exit (255);
; 4.  8: }
We don't need TOK_IDENT